Improve question block navigation

yscik opened the pull request:
Depends on  #3967

### Changes proposed in this Pull Request

* Automatically insert an empty question block at the end of the quiz
  — inserts a new empty block if the quiz block is selected and the last question is not empty (has a title)
  — removes auto-inserted empty block if the quiz block is not focused
* Jump to question description on Enter keypress in question title
* Delete question block on backspace in empty question title 
  — :question: Should the block not be deleted if it has content (description, answers)?
* Remove rich-text formatting from question title
  — I don't think we wanted to add this for now
* Ensure first question block is focused when adding a quiz

### Testing instructions

* Add a quiz block. Add questions
* Check the quiz block always has an empty question at the end when selected
--
* Type in a question title, press `Enter`
* Check the question description block is now selected
--
* Delete a question title and press one more `Backspace`
* Check that the question is removed
--
* Check that in the Course Outline block, auto-inserted lessons in module blocks continue to work as expected

<!--
Helpful tips for screenshots:
https://en.support.wordpress.com/make-a-screenshot/
-->
### Screenshot / Video

![QuickTime movie](https://user-images.githubusercontent.com/176949/107822306-2d678d00-6d7e-11eb-8ea3-5d928493f527.gif)

jom commented:
- Creating a new question with `+` focuses the input on the question description instead of the question title, which would make more sense to me. It also sometimes doesn't do this on an auto-inserted block.
![2021-02-15 2 01 35 pm](https://user-images.githubusercontent.com/68693/107955975-6417f000-6f96-11eb-9c57-e1493eb52f82.gif)
- Eventually, it might make sense to check for any kind of dirty state in the question (not just title) before deleting the last question. With the above issue, I was typing descriptions and then would lose them if I accidentally focused away.

yscik commented:
> Creating a new question with + focuses the input on the question description

Fixed in 768c4ea 

> Eventually, it might make sense to check for any kind of dirty state in the question (not just title) before deleting

I haven't found a good way to check for changes with inner blocks involved. For now, I've added a check to not remove it if it was selected, as a proxy to being interacted with.

jom commented:
Looks good so far. Are you going to move `SingleLineInput` in this PR?

yscik commented:
Right, forgot that once again. Should be good to go now.
